### PR TITLE
chore: update golangci-lint

### PR DESCRIPTION
**Pull request by crenshaw-dev**
Just a bit of cleanup/maintenance. 

The linter revealed a bug, so I wrote unit tests.

### CODE REVIEW

File: util/io/bytereadseeker.go [hunk 1]
```diff
@@ -33,6 +32,5 @@ func (f byteReadSeeker) Seek(offset int64, whence int) (int64, error) {
 	if offset < 0 || offset > int64(len(f.data)) {
 		return 0, &fs.PathError{Op: "seek", Err: fs.ErrInvalid}
 	}
-	f.offset = offset
```
Comment (alexmt) [inline util/io/bytereadseeker.go:36]: Ops, this looks like a real bug , is not it? I think we should change `byteReadSeeker` to `*byteReadSeeker` instead.

**Review by alexmt — CHANGES_REQUESTED** [review]

File: util/io/bytereadseeker.go (hunk 1 above)
Comment (crenshaw-dev) [reply util/io/bytereadseeker.go:36]: Definitely could be a real bug... will look at what calls these methods.

Comment (crenshaw-dev) [reply util/io/bytereadseeker.go:36]: I wonder if we've just been lucky up to this point. Maybe index.html is small enough that it all fits into one read.

Comment (alexmt) [reply util/io/bytereadseeker.go:36]: Yeah, I think Seek just never been called so far. But it might be called in future

**Review by alexmt — APPROVED** [review]
LGTM

**Comment by crenshaw-dev** [comment]
Cherry-picked onto 2.3.

**Comment by crenshaw-dev** [comment]
Cherry-picked onto 2.2.

**Pull request merged**